src/components/toast-notifications/toastNotifications.js

```diff
@@ -1,49 +1,75 @@
 'use strict';
 
 angular.module('openshiftCommonUI')
-  .directive('toastNotifications', function(NotificationsService, $timeout) {
+  .directive('toastNotifications', function(NotificationsService, $rootScope, $timeout) {
     return {
       restrict: 'E',
       scope: {},
       templateUrl: 'src/components/toast-notifications/toast-notifications.html',
       link: function($scope) {
-        $scope.notifications = NotificationsService.getNotifications();
+        $scope.notifications = [];
 
-        $scope.close = function(notification) {
+        // A notification is removed if it has hidden set and the user isn't
+        // currently hovering over it.
+        var isRemoved = function(notification) {
+          return notifications.hidden && !notification.isHover;
+        };
+
+        // Remove items that are now hidden to keep the array from growing
+        // indefinitely. We loop over the entire array each digest loop, even
+        // if everything is hidden, and any watch update triggers a new digest
+        // loop. If the array grows large, it can hurt performance.
+        var removeHiddenToasts = function() {
+          $scope.notifications = _.reject($scope.notifications, isRemoved);
+        };
+
+        var closeNotification = function(notification) {
+          notification.isHover = false;
```

Comment:
is there any chance of setHover getting fired and re-setting this flag while the animation is running?